src/components/FlowStep/index.tsx

```diff
@@ -0,0 +1,69 @@
+import { FC } from "react"
+import { Box, BoxProps, Icon, useMultiStyleConfig } from "@chakra-ui/react"
+import { BodyMd, BodySm, LabelLg, LabelSm, LabelXs } from "../Typography"
+import { IoArrowForwardCircleSharp } from "react-icons/all"
+import {
+  FlowStepSize,
+  FlowStepStatus,
+  FlowStepVariant,
+} from "../../theme/FlowStep"
+
+export interface FlowStepProps extends BoxProps {
+  status?: FlowStepStatus
+  variant?: FlowStepVariant
+  size?: FlowStepSize
+  preTitle: string
+  title: string
+  description?: string | JSX.Element
+  isDescriptionHidden?: boolean
+  isDescriptionArrowHidden?: boolean
+}
+
+export const FlowStep: FC<FlowStepProps> = (props) => {
```

Comment:
Tbh this component looks very similar to `CheckListItem` (and `CheckListGroup`). I'm not sure as of now if we could reuse it in this case, but I think we could create it similarly - so this component would keep as `FlowStep` and we could create a new one - `FlowStepGroup` - where we would keep all the logic behind the flow steps.

I think that in `FlowStepGroup` component we just just have a prop of active step and we can just assume that all the previous steps are completed and pass proper props to `FlowStep` components.

Thoughts? @r-czajkowski @georgeweiler

Reply:
I like this idea but it only works for "simple" flow steps that follow a set pattern. There are many cases where the individual flowsteps need their own unique props, so a group component to manage the state wouldn't be useful. For example, not every flow step in a group follows the same logic. Here's an example that has steps in "open" state, even when they are "complete" and also include some content between the flow steps: 

![Screen Shot 2022-07-27 at 10 30 55 AM](https://user-images.githubusercontent.com/15269424/181301092-765adbe7-8a6e-4115-90a3-dc279d3dadbf.png)


Maybe it is still worth building for the "simple" use case, but I think treating as an individual is fine because the designs include many variants of the flowstep component.

Reply:
design reference: https://www.figma.com/file/Tx1lZc7us4SqCeAgbt1hBV/Threshold-Dapp?node-id=3306%3A137922

Reply:
Hmmm, for such case we could maybe add another prop like `allActive: boolean` which will make all the steps extended and active regardless of the `active` prop. But I think for now it can stay as it is, and we will refactor this if there will be such a need.